gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/JWTFederationFilter.java

```diff
@@ -136,6 +154,22 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
       continueWithAnonymousSubject(request, response, chain);
       return;
     }
+
+    if (useCookie) {
+      try {
+        if (authenticateWithCookies((HttpServletRequest) request, (HttpServletResponse) response, chain)) {
+          // if there was a valid cookie authentication was handled, there is no point in
+          // going forward to check the JWT path in the header
+          return;
+        }
+      } catch (NoValidCookiesException e) {
+        log.missingValidCookie();
```

Comment:
This might be an issue in case request has invalid cookie but valid token. This could potentially be a regression.